packages/compiler/core/semantic-walker.ts

```diff
@@ -252,17 +253,30 @@ function navigateType(
   }
 }
 
-// Find the immediate children of the given model.
-export function findChildModels(program: Program, parent: ModelType) {
-  const children: ModelType[] = [];
+function isTemplate(model: ModelType) {
+  return (
+    model.node.kind === SyntaxKind.ModelStatement &&
+    model.node.templateParameters.length > 0 &&
+    !model.templateArguments?.length
+  );
+}
+
+// Produce a map from models in a program (with children) to their children. Model templates are not included.
+export function mapChildModels(program: Program): ReadonlyMap<ModelType, readonly ModelType[]> {
+  const map = new Map<ModelType, ModelType[]>();
   navigateProgram(program, {
     model: (model) => {
-      if (model.baseModel === parent) {
-        children.push(model);
+      if (model.baseModel && !isTemplate(model) && !isTemplate(model.baseModel)) {
```

Comment:
This means we can't do this? At least in the OpenAPI conversion?

```cadl
model Pet {
  name: string
}

model Cat<C> extends Pet {
   color: C;
}
```

Should then maybe this check be moved to the emitter?

Reply:
You can do that, but if you have say `op foo(): Pet`, we won't emit `Cat<C>`. We can't emit templates to OpenAPI. But you're right, I think it might make sense to move this exclusion of templates to the emitter and not have mapChildModels have an opinion here. That said, there was talk of having semantic-walker not visit templates by default? @bterlson any thoughts here?

Reply:
This becomes a seperate issue than this PR though. Maybe keeping it as it is for now and creating an issue for it.

Reply:
Sounds good. I added a checkbox to https://github.com/microsoft/cadl/issues/223